client/src/components/auth/RegistrationForm.js

```diff
@@ -11,139 +11,153 @@ import { createTeam } from '../../actions/team';
 import './Register.css';
 import { useDispatch, useSelector } from 'react-redux';
 import { useHistory } from 'react-router-dom';
-import toast from 'react-hot-toast';
+import { Formik } from 'formik';
+import * as yup from 'yup';
 
 export default function RegistrationForm() {
   const dispatch = useDispatch();
   const history = useHistory();
-  const [validated, setValidated] = useState(false);
   const isRegistered = useSelector(state => state.auth.isRegistered)
 
   useEffect(() => {
-    if(isRegistered && validated){
+    if(isRegistered){
       history.push('/login')
     }
-  }, [history, isRegistered, validated])
+  }, [history, isRegistered])
 
-  const [inputs, setInputs] = useState({
+  const teamInfoSchema = yup.object({
+    teamName: yup
+      .string()
+      .required('Team Name is required')
+      .min(3, 'Must be at least 3 characters'),
+    orgName: yup
+      .string()
+      .required('Organization Name is required')
+      .min(3, 'Must be at least 3 characters'),
+    email: yup
+      .string()
+      .email('Invalid Email')
+      .required('Email is required'),
```

Comment:
Can we improve this to have proper email domain validation? Not sure how complicated it'll get though

Reply:
If we can have password regex, we should be able to do it for email as well. The regex for email is widely documented online, e.g. on StackOverflow

Reply:
Maybe have a check to see if '@' is at least present, as a valid email.

Reply:
I think the built-in .email() function from formik already checking if '@' is present when validating the email.
In terms of checking the proper domain, I had a search but couldn't find any relevant regex. I don't think it's possible to do it cause a new email domain can be registered by anyone unless we want to be restricted the input email to a specific domain e.g. google.com.